File: Logic/BaseDM.cs
using Shared.Interfaces;
using System;

namespace Logic
{
    public class BaseDM : IDisposable
    {
        protected IFactory Factory { get; }

        public BaseDM(IFactory factory)
        {
            Factory = factory;
        }

        public void Dispose()
        {
            GC.SuppressFinalize(this);
        }
    }
}

File: LogicVB/BaseDM.vb
Imports [Shared].Interfaces

Public Class BaseDM
    Implements IDisposable

    Protected Property Factory As IFactory

    Public Sub New(factory As IFactory)
        Me.Factory = factory
    End Sub

    Public Sub Dispose() Implements IDisposable.Dispose
        GC.SuppressFinalize(Me)
    End Sub
End Class

